warp10/src/main/java/io/warp10/continuum/store/MultiScanGTSDecoderIterator.java

```diff
@@ -271,12 +257,8 @@ public boolean hasNext() {
       }
     } else {
       scan.setStartRow(startkey);
-      if (endAtMinlong) {
-        // When the end timestamp should have been after MIN_LONG add a trailing 0x00
-        scan.setStopRow(Arrays.copyOf(endkey, endkey.length + 1));
-      } else {
-        scan.setStopRow(endkey);
-      }
+      // Then endkey is 'endkey' plus a trailing 0x00 so we include 'endkey'
+      scan.setStopRow(Arrays.copyOf(endkey, endkey.length + 1));
```

Comment:
This line is OK, but the preBoundaryScan will also fetch `endkey`. I think `endkey` should always have a trailing 0x00.